torch/nested/_internal/ops.py

```diff
@@ -536,6 +569,52 @@ def is_same_size_default(func, *args, **kwargs):
     return args[0]._size == args[1]._size
 
 
+@register_jagged_func(
+    torch.ops.aten.sum.dim_IntList, "self: jt, dim: any?, keepdim: any?, dtype: any?"
+)
+def sum_dim_IntList(func, *args, **kwargs):
+    from torch._prims_common import canonicalize_dims
+
+    # sum_dim_IntList can produce a NT or a T depending on whether the ragged dim
+    # is summed over
+    _, new_kwargs = normalize_function(
+        func, args=args, kwargs=kwargs, normalize_to_only_use_kwargs=True
+    )
+    inp = new_kwargs.pop("input")
+    new_kwargs["dim"] = tuple(
+        canonicalize_dims(inp.dim(), s) for s in new_kwargs["dim"]
+    )
```

Comment:
nit: I've been using `_wrap_jagged_dim()` for canonicalization + conversion to jagged dim space (i.e. subtracting 1). maybe that can be used here for consistency as well? I think you'd need to tweak it some to avoid the restrictive dim check under certain cases.

Reply:
Good idea, I refactored this a little.